server/routes/v1/version.ts

```diff
@@ -138,3 +140,56 @@ export const updateLastViewed = [
     return res.json({ version: id, role })
   },
 ]
+
+export const deleteVersion = [
+  ensureUserRole('user'),
+  async (req: Request, res: Response) => {
+    const { id } = req.params
+    const { user } = req
+
+    const version = await findVersionById(user, id)
+
+    if (!version) {
+      throw NotFound({ code: 'version_not_found', id }, 'Unable to find version')
+    }
+
+    if (req.user.id !== version.metadata.contacts.uploader) {
+      throw Forbidden({ code: 'user_unauthorised' }, 'User is not authorised to do this operation.')
+    }
+
```

Comment:
Almost all of this logic and below should be in `services/` functions.  Routes handle formatting the request, business and response logic.  Services handle altering / searching the database.  This idea has several merits:

- Less duplication of logic (e.g. we'll want to delete versions when deleting models as well, so good to put that code somewhere accessible to both)
- More secure (most services require `req.user` which can be used for access control, etc).

Reply:
This was actually my plan to refactor this method once it's finished (still working on the registry bit). I was going to split it up into separate functions if I could, but it might all go in one service function, think it depends how large it's going to be.